pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -27,52 +27,50 @@ import (
 )
 
 var (
-	log                     = logger.Log
-	ResultNormal            = common.ResultNormal
-	ResultRequeue           = common.ResultRequeue
-	ResultRequeueAfter5mins = common.ResultRequeueAfter5mins
-	MetricResType           = common.MetricResTypeVPC
+	log           = logger.Log
+	MetricResType = common.MetricResTypeNetworkInfo
 )
 
-// VPCReconciler VPCReconcile reconciles a VPC object
-type VPCReconciler struct {
+// NetworkInfoReconciler NetworkInfoReconcile reconciles a NetworkInfo object
+// Actually it is more like a shell, which is used to manage nsx VPC
+type NetworkInfoReconciler struct {
 	Client   client.Client
 	Scheme   *apimachineryruntime.Scheme
 	Service  *vpc.VPCService
 	Recorder record.EventRecorder
 }
 
-func (r *VPCReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	obj := &v1alpha1.VPC{}
-	log.Info("reconciling VPC CR", "VPC", req.NamespacedName)
-	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, common.MetricResTypeVPC)
+func (r *NetworkInfoReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	obj := &v1alpha1.NetworkInfo{}
+	log.Info("reconciling NetworkInfo CR", "NetworkInfo", req.NamespacedName)
+	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerSyncTotal, common.MetricResTypeNetworkInfo)
 
 	if err := r.Client.Get(ctx, req.NamespacedName, obj); err != nil {
-		log.Error(err, "unable to fetch VPC CR", "req", req.NamespacedName)
+		log.Error(err, "unable to fetch NetworkInfo CR", "req", req.NamespacedName)
 		return common.ResultNormal, client.IgnoreNotFound(err)
 	}
 
 	if obj.ObjectMeta.DeletionTimestamp.IsZero() {
-		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateTotal, common.MetricResTypeVPC)
-		if !controllerutil.ContainsFinalizer(obj, commonservice.VPCFinalizerName) {
-			controllerutil.AddFinalizer(obj, commonservice.VPCFinalizerName)
+		metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerUpdateTotal, common.MetricResTypeNetworkInfo)
```

Comment:
Do we need NetworkInfo finalizer? Or we could just delete VPC when NS is deleted

Reply:
We should have it if we delete NetworkInfo directly.